packages/core/src/config/load.ts

```diff
@@ -102,3 +108,21 @@ export async function getConfig(configPath: string | undefined = findConfig()):
     throw new Error(`Error parsing config file at '${configPath}': ${e.message}`);
   }
 }
+
+interface CreateConfigOptions {
+  extends?: string[];
+  tokens?: RegionalTokenWithValidity[],
+}
+
+export async function createConfig (content: string, options?: CreateConfigOptions): Promise<Config>;
+export async function createConfig (rawConfig: RawConfig, options?: CreateConfigOptions): Promise<Config>;
+export async function createConfig (rawConfig: any, options: CreateConfigOptions = {}): Promise<Config> {
+  if (typeof rawConfig === 'string') {
+    rawConfig = parseYaml(rawConfig);
+  }
+
+  return addConfigMetadata({
+    rawConfig: transformConfig(rawConfig),
+    ...options
+  });
+}
```

Comment:
I'd avoid overloading.

Maybe union types would work better?

```ts
export async function createConfig (rawConfig: rawConfig | string, options: CreateConfigOptions = {}): Promise<Config> {
 ...
}


```

It's shorter and it's JavaScript.

Reply:
agree

Reply:
Changed